lib/sshutils/server.go

```diff
@@ -512,16 +512,27 @@ func (s *Server) HandleConnection(conn net.Conn) {
 			s.log.Warnf("Dropping inbound ssh connection due to error: %v", err)
 			// Immediately dropping the ssh connection results in an
 			// EOF error for the client.  We therefore wait briefly
-			// to see if the client opens a channel, which will give
-			// us the opportunity to respond with a human-readable
-			// error.
-			select {
-			case firstChan := <-chans:
-				if firstChan != nil {
-					firstChan.Reject(ssh.Prohibited, err.Error())
+			// to see if the client opens a channel or sends any global
+			// requests, which will give us the opportunity to respond
+			// with a human-readable error.
+			for wait := true; wait; {
```

Comment:
What's the point of the loop and the `wait`? Why do we keep waiting only after a global request?

Reply:
This would read way better with an infinite loop and `break`s/`continue`s IMO.